exp/controllers/azuremangedcontrolplane_controller.go

```diff
@@ -161,8 +163,29 @@ func (r *AzureManagedControlPlaneReconciler) Reconcile(ctx context.Context, req
 
 	log = log.WithValues("machinePool", ownerPool.Name)
 
+	// check if the control plane's namespace is allowed for this identity and update owner references for the identity.
```

Comment:
why are you updating the Owner reference of the identity? what if this identity is used by two different clusters?

Reply:
We update the ownerReferences for `AzureClusterIdentity` which has 1:1 relationship with the cluster. We do similarly in [azure cluster controller](https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/83853b7148355637cd8329e9edf64b67ffa9af79/controllers/azurecluster_controller.go#L211).

Reply:
yeah, you are right, my bad!